frontend/catalyst/third_party/oqc/src/CMakeLists.txt

```diff
@@ -10,18 +10,37 @@ set(oqc_backend_dir "${OQC_BUILD_DIR}/backend")
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# Locate Python & PyBind11
-# The optional component is only used for the C++ test suite (to spin up their its interpreter),
-# and requires libpython.so to be available on the system.
-find_package(Python REQUIRED
-    COMPONENTS Interpreter Development.Module
-    OPTIONAL_COMPONENTS Development.Embed
-)
+set(PYTHON_VERSION_TO_FIND "3" CACHE STRING "Python version to find and use")
+
+find_package(Python ${PYTHON_VERSION_TO_FIND} EXACT
+    REQUIRED COMPONENTS Interpreter Development.Module
+    OPTIONAL_COMPONENTS Development.SABIModule)
+
```

Comment:
This file looks good except maybe this first bit I think. For reference in this recent PR we've overhauled and streamlined everything related to python & python package discovery: #1380 

The key aspects are:
- specify the `Python_EXECUTABLE` variable on the CMake command line which will take care of finding the right environment during the `find_package(Python)` call (no other variables, like `PYTHON_VERSION_TO_FIND` etc, should be needed)
- specify `Interpreter` and `Development.Module` components for building a python module
- specify `Development.Embed` components for building a standalone binary that can loaded/run outside of Python (this requires pybind11), make this component optional if there are targets that don't rely on interpreter embedding
- invoke `Python_EXECUTABLE` from within cmake to find the relevant packages

Reply:
Removed